File: src/main/java/com/appinbox/sdk/msg/DetailFragment.java
package com.appinbox.sdk.msg;

import android.app.Activity;
import android.content.SharedPreferences;
import android.os.Bundle;

import com.appinbox.sdk.R;
import com.appinbox.sdk.svc.ApiBuilder;
import com.google.android.material.appbar.CollapsingToolbarLayout;

import androidx.fragment.app.Fragment;

import android.os.Handler;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

import android.content.Context;

/**
 * A fragment representing a single Message detail screen.
 * This fragment is either contained in a {@link ListActivity}
 * in two-pane mode (on tablets) or a {@link DetailActivity}
 * on handsets.
 */
public class DetailFragment extends Fragment {
    /**
     * The fragment argument representing the item ID that this fragment
     * represents.
     */
    public static final String ARG_ITEM_ID = "item_id";
    public static final String MSG_KEY = "msg";
    private String appId;
    private String appKey;
    private String contact;

    /**
     * The dummy content this fragment is presenting.
     */
    private Message mItem;

    /**
     * Mandatory empty constructor for the fragment manager to instantiate the
     * fragment (e.g. upon screen orientation changes).
     */
    public DetailFragment() {
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if (getArguments().containsKey(MSG_KEY)) {
            // Load the dummy content specified by the fragment
            // arguments. In a real-world scenario, use a Loader
            // to load content from a content provider.
            mItem = getArguments().getParcelable(MSG_KEY);

            Activity activity = this.getActivity();
            CollapsingToolbarLayout appBarLayout = activity.findViewById(R.id.toolbar_layout);
            if (appBarLayout != null) {
                appBarLayout.setTitle(mItem.title);
            }
        }
        loadPref();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.f_message_detail, container, false);

        // Show the dummy content as text in a TextView.
        if (mItem != null) {
            ((TextView) rootView.findViewById(R.id.message_detail)).setText(mItem.body);
        }
        new Handler().postDelayed(() -> ApiBuilder.getInstance().readMessage(appId, appKey, contact, mItem.id).enqueue(new Callback<Message>() {
            @Override
            public void onResponse(Call<Message> call, Response<Message> response) {

            }

            @Override
            public void onFailure(Call<Message> call, Throwable t) {

            }
        }), 5000);

        return rootView;
    }

    private void loadPref() {
        SharedPreferences preferences = this.getActivity().getSharedPreferences(getString(R.string.preference_file_key), Context.MODE_PRIVATE);
        appId = preferences.getString(getString(R.string.sp_app), "");
        appKey = preferences.getString(getString(R.string.sp_key), "");
        contact = preferences.getString(getString(R.string.sp_contact), "");
    }
}

File: src/main/java/com/appinbox/sdk/ui/DetailFragment.kt
package com.appinbox.sdk.ui

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.fragment.navArgs
import com.appinbox.sdk.databinding.FDetailBinding
import com.appinbox.sdk.repo.svc.ApiBuilder

/**
 * A simple [Fragment] subclass as the second destination in the navigation.
 */
class DetailFragment : Fragment() {

    private var _binding: FDetailBinding? = null

    // This property is only valid between onCreateView and
    // onDestroyView.
    private val binding get() = _binding!!
    private val args: DetailFragmentArgs by navArgs()


    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {

        _binding = FDetailBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.tvDetailTitle.text = args.title
        binding.tvDetailBody.text = args.body
        binding.tvDetailSentAt.text = args.sentAt
        binding.tvDetailReadAt.text = args.readAt
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}